src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/PublicSymbols/AnonymousType.DelegatePublicSymbol.cs

```diff
@@ -28,10 +30,9 @@ internal override NamedTypeSymbol MapToImplementationSymbol()
 
             internal override AnonymousTypeOrDelegatePublicSymbol SubstituteTypes(AbstractTypeMap map)
             {
-                var typeDescr = TypeDescriptor.SubstituteTypes(map, out bool changed);
-                return changed ?
```

Comment:
>changed ? [](http://example.com/codeflow?start=23&length=9)

It is not clear why the optimization is dropped. I think ```SubstituteTypes``` is supposed to return the original instance if substitution has no impact.

Reply:
This was from an earlier iteration and should have been reverted. Thanks.